bftengine/src/bftengine/ReplicaImp.cpp

```diff
@@ -5798,66 +5782,9 @@ void ReplicaImp::executeNextCommittedRequests(concordUtils::SpanWrapper &parent_
-      SigManager::instance()->sign(reinterpret_cast<char *>(data_vec.data()), data_vec.size(), sig.data());
-      req.signature = std::vector<uint8_t>(sig.begin(), sig.end());
-      data_vec.clear();
-      concord::messages::serialize(data_vec, req);
-      std::string strMsg(data_vec.begin(), data_vec.end());
-      auto requestSeqNum =
-          std::chrono::duration_cast<std::chrono::microseconds>(getMonotonicTime().time_since_epoch()).count();
-      auto crm = new ClientRequestMsg(internalBFTClient_->getClientId(),
-                                      RECONFIG_FLAG,
-                                      requestSeqNum,
-                                      strMsg.size(),
-                                      strMsg.c_str(),
-                                      60000,
-                                      "wedge-noop-command-" + std::to_string(lastExecutedSeqNum + 1));
-      // Now, try to send a new PrePrepare message immediately, without waiting to a new batch
-      onMessage(std::unique_ptr<ClientRequestMsg>(crm));
-      tryToSendPrePrepareMsg(false);
-    }
-  }
-  auto seqNumToStopAt = ControlStateManager::instance().getCheckpointToStopAt();
-  if (seqNumToStopAt.has_value() && seqNumToStopAt.value() > lastExecutedSeqNum && isCurrentPrimary()) {
-    // If after execution, we discover that we need to wedge at some future point, push a noop command to the incoming
-    // messages queue.
-    LOG_INFO(GL, "sending noop command to bring the system into wedge checkpoint");
-    concord::messages::ReconfigurationRequest req;
-    req.sender = config_.replicaId;
-    req.command = concord::messages::WedgeCommand{config_.replicaId, true};
-    // Mark this request as an internal one
-    std::vector<uint8_t> data_vec;
-    concord::messages::serialize(data_vec, req);
-    std::string sig(SigManager::instance()->getMySigLength(), '\0');
-    SigManager::instance()->sign(reinterpret_cast<char *>(data_vec.data()), data_vec.size(), sig.data());
-    req.signature = std::vector<uint8_t>(sig.begin(), sig.end());
-    data_vec.clear();
-    concord::messages::serialize(data_vec, req);
-    std::string strMsg(data_vec.begin(), data_vec.end());
-    auto current_time =
-        std::chrono::duration_cast<std::chrono::microseconds>(getMonotonicTime().time_since_epoch()).count();
-    auto crm = new ClientRequestMsg(internalBFTClient_->getClientId(),
-                                    RECONFIG_FLAG,
-                                    current_time,
-                                    strMsg.size(),
-                                    strMsg.c_str(),
-                                    60000,
-                                    "wedge-noop-command-" + std::to_string(lastExecutedSeqNum));
-    // Now, try to send a new PrePrepare message immediately, without waiting to a new batch
-    onMessage(std::unique_ptr<ClientRequestMsg>(crm));
-    tryToSendPrePrepareMsg(false);
+    primaryPushNoOpIfWedgePending(lastExecutedSeqNum + 1);
```

Comment:
shouldn't it be lastExecutedSeqNum (without the +1)?

Reply:
the original version had the +1